src/main/java/mate/academy/onlinebookstore/repository/book/BookRepository.java

```diff
@@ -1,8 +1,15 @@
 package mate.academy.onlinebookstore.repository.book;
 
+import java.util.List;
 import mate.academy.onlinebookstore.model.Book;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.EntityGraph;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.data.jpa.repository.Query;
 
 public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {
+    @EntityGraph(attributePaths = "categories")
+    @Query("SELECT b FROM Book b JOIN b.categories c WHERE c.id = :categoryId")
```

Comment:
```suggestion
    @EntityGraph(attributePaths = "categories")
```

Reply:
it doesn't work without this line

Reply:
I meant that you can leave only one of these annotaitons
`@EntityGraph(attributePaths = "categories")` preferable beacause it improve redability